packages/deeplinks/observability/locators/observability_onboarding.ts

```diff
@@ -12,4 +12,5 @@ export const OBSERVABILITY_ONBOARDING_LOCATOR = 'OBSERVABILITY_ONBOARDING_LOCATO
 export interface ObservabilityOnboardingLocatorParams extends SerializableRecord {
   /** If given, it will load the given map else will load the create a new map page. */
   source?: 'customLogs' | 'systemLogs';
+  query?: string;
```

Comment:
**suggestion:** what about passing separate type parameters instead of a query string? This allows for passing anything and doesn't help when using the locator in passing the allowed parameters, which would make the experience consistent with the `source` parameter definition.

The parameters that should compose the URL query string can then be composed together in the locator `getLocation` definition using the URLSearchParams API
```ts
const queryString = new URLSearchParams({category: 'infra', foo: 'bar'}).toString()
// category=infra&foo=bar
```

Reply:
I actually started to do this but worried that it was too pedantic and chose the simpler implementation. I can certainly add this.

Reply:
Update: done dab499aee98f5fd41959c52f10a967cfd7bba5a2